### DISCUSSION TITLE
Receive a signal in C# code

### DISCUSSION THREAD

**Discussion by vabs212** [elsa-workflows/elsa-core#1442] [Design & tech]
I am snedin a Signal in C# code which i sent in an activity using the below code 

         if(Userdetails.SendSms)
            {
                Signal = "ValidationSmsSignal";                
                Input = "ValidationSmsSignal";
               // context.Input = Input;
                await _signaler.TriggerSignalAsync(Signal, Input,  correlationId: CorrelationId, cancellationToken: context.CancellationToken);
               return Done();
            }
is there something wrong .if not then please someone tell me how can i receive the Signal in C# only .
Please help

**Comment by sfmskywalker** [comment]
There doesn't seem to be anything wrong with your code, but it's not clear what the issue is. Can you describe what you want to do, what you tried and what you expect to happen? Then I'm sure we can help you solve it.

**Reply by vabs212** [reply]
actually i am doing a POC on elsa in which i am creating a user and storing the user data in DB. After that on user input i am choosing how user want to communicate either through SMS or API . 
Below is the Create User Activity that i created. I am receiving data through Post API in the below Activity. 

![image](https://user-images.githubusercontent.com/66916559/130960246-40b37aad-4284-4b44-8b84-07aa0cd88ec9.png)

After that i am implementing fork through Dashboard .from there i have two branches and on the basis of parameters i want to execute one or both given on the basis of parameters .so put send and receive  signals on both branch.
Code for sending the signals are as follows .i Created a custom  send signal activity and custome receive signal activity whose code is below(just changed the input in C# not taking it from dashboard.)
Below is the custom code for sending trigger

![image](https://user-images.githubusercontent.com/66916559/130985771-e9c60262-a09c-4111-bc6d-46236b918743.png)

Below is the code for receiving trigger (copied from your code)
![image](https://user-images.githubusercontent.com/66916559/130986053-1c896b94-589c-48fe-8cdf-8c1ff70b15a7.png)

Now in the above code when the signal comes in receive trigger  
in this line 
1)context.WorkflowExecutionContext.IsFirstPass ? OnResume(context) : Suspend()
always getting firstpass as false
2)Sometimes getting input as null  sometimes getting input as post data of the first activity that is used to start the workflow 
whixh result in suspending the consecutive activity which i am not able to.
My flow diagarm is 
Sample API parameters
{
"Name":"Vaibhav Sharma1",
"Email":"vaibhav.sharma@easyrewardz.com",
"Password":"India@5551",
"RepeatPassword":"India@5551",
"SendMail":true,
"SendSms":false
}

 
![image](https://user-images.githubusercontent.com/66916559/130987278-01a5578c-d9d6-4b6f-b645-8e0d52f28416.png)

Startup:
![image](https://user-images.githubusercontent.com/66916559/130987684-8b58c3b9-036f-42cc-b51b-5bbf2e0014bf.png)


Am i missing something in startup or what??
Please help

**Reply by sfmskywalker** [reply]
I'm having a hard time putting everything together due to partial code and the screenshot being too small. Is there any way you can share your POC project with me so I can run it for myself?

**Reply by vabs212** [reply]
i can share my project with you.please provide me your mail ID or something

**Reply by sfmskywalker** [reply]
Send it to me via Discord in a DM.
